pkg/iam/route53_template.go

```diff
@@ -24,19 +24,24 @@ const route53TrustIdentityPolicyWithIRSA = `{
       "Action": "sts:AssumeRole"
     },
     {
-        "Effect": "Allow",
-        "Principal": {
-            "Federated": "arn:aws:iam::{{.AccountID}}:oidc-provider/{{.CloudFrontDomain}}"
-        },
-        "Action": "sts:AssumeRoleWithWebIdentity",
-        "Condition": {
-            "StringEquals": {
-                "{{.CloudFrontDomain}}:sub": "system:serviceaccount:{{.Namespace}}:{{.ServiceAccount}}"
-            }
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::{{.AccountID}}:oidc-provider/{{.CloudFrontDomain}}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "{{.CloudFrontDomain}}:sub": [{{- range $index, $sa :=  .ServiceAccounts -}}
+            {{- if $index }}, {{ end -}}
+              "system:serviceaccount:{{$sa.Namespace}}:{{$sa.Name}}"
+            {{- end -}}
+          ]
```

Comment:
I think we need to change that slightly. According to the information I have we would need to use `ForAnyValues:StringEquals` to add multiple service accounts:

Here's an example how it would look like (taken from [CAPI docs](https://cluster-api-aws.sigs.k8s.io/topics/specify-management-iam-role.html#iam-role-trust-policy)):
```
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "ForAnyValue:StringEquals": {
          "oidc.eks.${AWS_REGION}.amazonaws.com/id/${OIDC_PROVIDER_ID}:sub": [
            "system:serviceaccount:capa-system:capa-controller-manager",
            "system:serviceaccount:capi-system:capi-controller-manager",
            "system:serviceaccount:capa-eks-control-plane-system:capa-eks-control-plane-controller-manager",
            "system:serviceaccount:capa-eks-bootstrap-system:capa-eks-bootstrap-controller-manager",
          ]
        }

Reply:
👊🏾